src/expression.d

```diff
@@ -14695,7 +14694,7 @@ extern (C++) final class PowExp : BinExp
         }
         e = new ScopeExp(loc, mmath);
 
-        if (e2.op == TOKfloat64 && e2.toReal() == 0.5)
+        if (e2.op == TOKfloat64 && e2.toReal() == real_t(0.5))
```

Comment:
Scattered around the place are this, `real_t(1)`, `real_t(-1)`, and `real_t(0)`.  As these are so common shouldn't they instead be properties of `CTFloat`?  E.g: `CTFloat.zero`, `CTFloat.one`, `CTFloat.minusone`, `CTFloat.half`.

Having it in a cached constant would be better than having to support converting from native double or int to real_t.  Bonus motivation for GDC, all these numbers are constants in GCC's real implementation.  :-)